tests/functional/artifacts/test_previous_version_state.py

```diff
@@ -82,7 +300,11 @@ def compare_previous_state(
         ]
         if expect_pass:
             results = run_dbt(cli_args, expect_pass=expect_pass)
-            assert len(results) == 0
+            # We have different numbers of nodes in older versions
+            if compare_manifest_version < 7:
+                assert len(results) == 5
```

Comment:
I like making the version differences really clear and obvious; with an expected_number_of results variable it would get buried in the ternary expression.

Reply:
Actually, on second thought, we really don't care about the number of results here, since we're just testing that the manifest can be loaded when doing state:modified, so I changed it to always be len(results) == 0